Test failure in PyICU 1.9.3

I am getting one test failure with the latest pyicu 1.9.3 release:

```
======================================================================
FAIL: testGetBestPattern (test.test_DateTimeParserGenerator.TestDateTimePatternGenerator)
Test a few different languages and common patterns.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/python-pyicu/src/PyICU-1.9.3/test/test_DateTimeParserGenerator.py", line 74, in testGetBestPattern
    self.assertEqual(sdf.format(self.date), locale_data[index])
AssertionError: 'May 10' != 'May 9'
- May 10
?     ^^
+ May 9
?     ^


----------------------------------------------------------------------
Ran 38 tests in 0.339s

FAILED (failures=1)
```

Any hints?

There is probably still a bug that depends on your local timezone (where are you located ? what is your default timezone ?)
A fix would be welcome !

Andi..

> On May 27, 2016, at 23:42, Felix Yan <notifications@github.com> wrote:
> 
> I am getting one test failure with the latest pyicu 1.9.3 release:
> 
> ======================================================================
> FAIL: testGetBestPattern (test.test_DateTimeParserGenerator.TestDateTimePatternGenerator)
> Test a few different languages and common patterns.
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/build/python-pyicu/src/PyICU-1.9.3/test/test_DateTimeParserGenerator.py", line 74, in testGetBestPattern
>     self.assertEqual(sdf.format(self.date), locale_data[index])
> AssertionError: 'May 10' != 'May 9'
> - May 10
> ?     ^^
> + May 9
> ?     ^
> 
> 
> ----------------------------------------------------------------------
> Ran 38 tests in 0.339s
> 
> FAILED (failures=1)
> Any hints?
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

--- a/test/test_DateTimeParserGenerator.py
+++ b/test/test_DateTimeParserGenerator.py
@@ -71,6 +71,7 @@ class TestDateTimePatternGenerator(TestCase):
             for index, skeleton in enumerate(skeletons):
                 pattern = dtpg.getBestPattern(skeleton)
                 sdf = SimpleDateFormat(pattern, locale)
+                sdf.setTimeZone(self.tz)
                 self.assertEqual(sdf.format(self.date), locale_data[index])
 
     def testReplaceFieldType(self):

